demos/sample_azure_iot_ca_recovery/sample_azure_iot_ca_recovery.c

```diff
@@ -605,28 +613,36 @@ static uint32_t prvConnectToServerWithBackoffRetries( const char * pcHostName,
 
         if( xNetworkStatus != eTLSTransportSuccess )
         {
-            /* Generate a random number and calculate backoff value (in milliseconds) for
-             * the next connection retry.
-             * Note: It is recommended to seed the random number generator with a device-specific
-             * entropy source so that possibility of multiple devices retrying failed network operations
-             * at similar intervals can be avoided. */
-            xBackoffAlgStatus = BackoffAlgorithm_GetNextBackoff( &xReconnectParams, configRAND32(), &usNextRetryBackOff );
-
-            if( xBackoffAlgStatus == BackoffAlgorithmRetriesExhausted )
+            if( xNetworkStatus == eTLSTransportCAVerifyFailed )
```

Comment:
Nit: in the final version, the code should retry several times (configurable, using the default retry policy). 

We don't want this to happen too often. There are benign reasons a device might hit an invalid CA:
- Captive portal WiFi authentication (DNS gets redirected to the portal).
- Incorrect DNS configuration
- Localized, short-timed attacks.

Reply:
@wduraes brought up something similar. I can make a TODO here to circle back.